internal/provider/data_source_catalystcenter_network_devices_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Mozilla Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://mozilla.org/MPL/2.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccDataSourceCcNetworkDevices(t *testing.T) {
+	var checks []resource.TestCheckFunc
+	checks = append(checks, resource.TestCheckResourceAttr("data.catalystcenter_network_devices.test", "response.0.id", "e0ba1a00-b69b-45aa-8c13-4cdfb59afe65"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.catalystcenter_network_devices.test", "response.0.hostname", "sw2.ciscotest.com"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.catalystcenter_network_devices.test", "response.0.management_ip_address", "10.10.10.1"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.catalystcenter_network_devices.test", "response.0.platform_id", "C9KV-UADP-8P"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.catalystcenter_network_devices.test", "response.0.role", "CORE"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.catalystcenter_network_devices.test", "response.0.software_type", "IOS-XE"))
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceCcNetworkDevicesConfig(),
+				Check:  resource.ComposeTestCheckFunc(checks...),
+			},
+		},
+	})
+}
+
+func testAccDataSourceCcNetworkDevicesConfig() string {
+	config := `resource "catalystcenter_network_devices" "test" {` + "\n"
```

Comment:
This would need to be updated as this resource does not exist. We would either need to add `test_prerequisites` to add a device first or assume a certain device already exists. We can keep it like this for now and I would then update it later to work with our CI environment.

Reply:
Yes, let's add a test tag for now and we will see how we can update the test to work in our ci environment.

Reply:
done - tag `INVENTORY`